programs/ziti-edge-tunnel/ziti-edge-tunnel.c

```diff
@@ -1818,8 +1823,13 @@ static void run(int argc, char *argv[]) {
             mask |= (ip[i] & 0xFFU);
         }
 
-        tun_ip = htonl(mask | 0x1);
-        dns_ip = htonl(mask | 0x2);
+        tun_ip = htonl(mask);
+        dns_ip = htonl(mask + 1);
+        uint32_t dns_nw_addr = mask & (0xFFFFFFFFUL << (32 - bits)) & 0xFFFFFFFFUL;
+
+        ip_addr_t dns_ip4_addr = IPADDR4_INIT(htonl(dns_nw_addr));
+        dns_range = strdup(ipaddr_ntoa(&dns_ip4_addr));
+        sprintf(dns_range, "%s/%d", strdup(ipaddr_ntoa(&dns_ip4_addr)), bits);
```

Comment:
make a new char* representing the network address of the "dns range" supplied. this allows one to supply the tun ip and subsequent DNS range in the same field. that's how it works currently but was flawed with the bitwise or'ing going on. instead above we use simple addition. we also do not expect a network address supplied. this is closer to how the zdew works as well.

Reply:
btw the inner `strdup` is unnecessary, and a leak. `ipaddr_ntoa` returns a static buffer, but it's ok because sprintf copies its contents into `dns_range`.

Reply:
alternatively, you could use `ipaddr_ntoa_r` to render the ip string directly into `dns_range` (which still needs to be alloc'ed a little larger), and then sprintf the prefix into `&dns_range[strlen(dns_range)]`.